plugins/reach/lib/kReachManager.php

```diff
@@ -474,13 +495,32 @@ private function updateReachProfileCreditUsage(EntryVendorTask $entryVendorTask)
 
 	private function handleErrorTask(EntryVendorTask $entryVendorTask)
 	{
+		if ($entryVendorTask->isScheduled())
+		{
+			$entryVendorTask->removeSchedulingData();
+		}
+
 		//Refund credit for tasks which could not be handled by the service provider
-		ReachProfilePeer::updateUsedCredit($entryVendorTask->getReachProfileId(), -$entryVendorTask->getPrice());
-		
-		//Rest task price so that reports will be aligned with the total used credit
-		$entryVendorTask->setOldPrice($entryVendorTask->getPrice());
-		$entryVendorTask->setPrice(0);
-		$entryVendorTask->save();
+		kReachUtils::refundTask($entryVendorTask);
+	}
+
+	private function handleAbortTask(EntryVendorTask $entryVendorTask)
```

Comment:
protected

Reply:
That would be incosistent with the pre existing handleErrorTask. do you want me to change it anyway?

Reply:
no, if thats the case keep it